sdk/identity/azure-identity/src/main/java/com/azure/identity/credential/ChainedTokenCredentialBuilder.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.credential;
+
+import com.azure.core.credentials.TokenCredential;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+/**
+ * Fluent credential builder for instantiating a {@link ChainedTokenCredential}.
+ *
+ * @see ChainedTokenCredential
+ */
+public class ChainedTokenCredentialBuilder {
+    private final Deque<TokenCredential> credentials;
+
+    /**
+     * Creates an instance of the builder to config the credential.
+     */
+    public ChainedTokenCredentialBuilder() {
+        this.credentials = new ArrayDeque<>();
+    }
+
+    /**
+     * Adds a credential to try to authenticate at the front of the chain.
+     * @param credential the credential to be added to the front of chain
+     * @return the ChainedTokenCredential itself
+     */
+    public ChainedTokenCredentialBuilder addFirst(TokenCredential credential) {
+        credentials.addFirst(credential);
+        return this;
+    }
+
+    /**
+     * Adds a credential to try to authenticate at the last of the chain.
+     * @param credential the credential to be added to the end of chain
+     * @return the ChainedTokenCredential itself
+     */
+    public ChainedTokenCredentialBuilder addLast(TokenCredential credential) {
+        credentials.addLast(credential);
+        return this;
+    }
+
+    /**
+     * @return a {@link ChainedTokenCredential} with the current configurations.
+     */
+    public ChainedTokenCredential build() {
+        return new ChainedTokenCredential(credentials);
```

Comment:
You should consider making a copy of credentials here, otherwise it may get modified afterwards if the user continues to user the builder and modify the credentials.